internal/pkg/istio-router-check/cmd/root.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/envoy"
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/helpers"
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/util/protomarshal"
+)
+
+const envoyRouterCheckTool string = "router_check_tool"
+
+type RootCommand struct {
+	ConfigDir               string
+	TestDir                 string
+	Details                 bool
+	DisableDeprecationCheck bool
+	OnlyShowFailures        bool
+	FailThreshold           float64
+	CoverageAll             bool
+	OutputDir               string
+	DetailedCoverage        bool
+	Verbosity               int
+}
+
+func NewCmdRoot() (*cobra.Command, error) {
+	rootCmd := &RootCommand{}
+	cmd := &cobra.Command{
+		Use: "istio-router-check",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := exec.LookPath(envoyRouterCheckTool); err != nil {
+				return fmt.Errorf("missing route table check tool: %w", err)
+			}
+			logger := slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+				Level: slog.Level(rootCmd.Verbosity),
+			}))
+			ctx := logr.NewContextWithSlogLogger(cmd.Context(), logger)
+			cmd.SetContext(ctx)
+			return nil
+		},
+		RunE:         rootCmd.Run,
+		SilenceUsage: true,
+	}
+	cmd.Flags().StringVarP(&rootCmd.ConfigDir, "config-dir", "c", "", "directory containing virtualservices")
+	cmd.Flags().StringVarP(&rootCmd.TestDir, "test-dir", "t", "", "directory containing tests")
+	cmd.Flags().BoolVarP(&rootCmd.Details, "details", "d", true, "print detailed information about the test results")
+	cmd.Flags().BoolVarP(&rootCmd.DisableDeprecationCheck, "disable-deprecation-check", "", true, "disable deprecation check")
+	cmd.Flags().BoolVarP(&rootCmd.OnlyShowFailures, "only-show-failures", "", false, "only show failures")
+	cmd.Flags().Float64VarP(&rootCmd.FailThreshold, "fail-threshold", "f", 0.0, "threshold for failure")
+	cmd.Flags().BoolVarP(&rootCmd.CoverageAll, "covall", "", false, "measure coverage by checking all route fields")
+	cmd.Flags().StringVarP(&rootCmd.OutputDir, "output-dir", "o", "", "output directory for coverage information")
+	cmd.Flags().BoolVarP(&rootCmd.DetailedCoverage, "detailed-coverage", "", false, "print detailed coverage information")
+	cmd.Flags().IntVarP(&rootCmd.Verbosity, "", "v", 1, "log verbosity level")
+
+	requiredFlags := []string{"config-dir", "test-dir"}
+	for _, flag := range requiredFlags {
+		if err := cmd.MarkFlagRequired(flag); err != nil {
+			return nil, fmt.Errorf("failed to mark flag %q required: %w", flag, err)
+		}
+	}
+
+	return cmd, nil
+}
+
+func (c *RootCommand) Run(cmd *cobra.Command, _ []string) error {
```

Comment:
nit: We could decouple this from cobra a little more if we had `Run(ctx context.Context) error` and then called that from the command. Would make things more testable in the long term (and avoid having to use `cmd.Context()`.